src/types/boolean8.rs

```diff
@@ -0,0 +1,42 @@
+use std::cmp::Ordering;
+use std::convert::TryInto;
+
+use crate::compressor::Compressor;
+use crate::decompressor::Decompressor;
+use crate::types::{DataType, NumberLike};
+
+impl NumberLike for u8 {
```

Comment:
Let's really make this for `bool` instead of `u8`. `bytes_from` and `from_bytes` can still use a whole byte though